api/api/models/media.py

```diff
@@ -361,6 +363,22 @@ class AbstractMediaDecisionThrough(models.Model):
     class Meta:
         abstract = True
 
+    def perform_action(self):
+        """Perform the action specified in the decision."""
+
+        if self.decision.action in {
+            DecisionAction.DEINDEXED_SENSITIVE,
+            DecisionAction.DEINDEXED_COPYRIGHT,
+        }:
+            self.deleted_media_class.objects.create(media_obj=self.media_obj)
+
+        if self.decision.action == DecisionAction.MARKED_SENSITIVE:
+            self.sensitive_media_class.objects.create(media_obj=self.media_obj)
```

Comment:
Using `self.media_obj` will query the database, and isn't necessary here. (`_perform_es_action` queries it anyway to get the database identifier [truly an absurd and confusing distinction I will never understand why we don't do everything we can to get rid of], but we don't need to repeat bad patterns/habits and there it's unavoidable, unlike here where we can easily avoid it.)

```suggestion
        if self.decision.action in {
            DecisionAction.DEINDEXED_SENSITIVE,
            DecisionAction.DEINDEXED_COPYRIGHT,
        }:
            self.deleted_media_class.objects.create(media_obj_id=self.media_obj_id)

        if self.decision.action == DecisionAction.MARKED_SENSITIVE:
            self.sensitive_media_class.objects.create(media_obj_id=self.media_obj_id)
```

Reply:
Updated in fb8495f.